src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -301,7 +301,7 @@ private function processAnonymousServices(\DOMDocument $xml, $file)
         if (false !== $nodes = $xpath->query('//container:argument[@type="service"][not(@id)]|//container:property[@type="service"][not(@id)]')) {
             foreach ($nodes as $node) {
                 // give it a unique name
-                $id = sprintf('%s_%d', hash('sha256', $file), ++$count);
+                $id = sprintf('%d_%s', ++$count, hash('sha256', $file));
```

Comment:
just made these changes so that anonymous classes won't have their class set to a random string (+ added corresponding regexp check in ResolveClassPass)